Updating from manyrandom to fillaccounts

idiom-bytes opened the pull request:
manyrandom seems to be failing when trying to transfer tokens to users.
modified function (since it's ganache only) to mint tokens to other users (as designed by SimpleToken)
using it to easily distribute ocean to ganache pks

trentmc commented:
This PR is 7 days old. Is it still WIP or .. ?

Also: This PR doesn't seem to be attached to a github issue. Please one, and point this PR to it.

idiom-bytes commented:
Sorry, I have been stabilizing df-web before moving back to df-py to capture this work.

As you mentioned, I'm going to add tests for these new commands so they remain stable.

idiom-bytes commented:
I have:
- updated the dftool functions to better represent the latest patterns
- added tests to verify dftool functions are working
- verified tests are passing locally

idiom-bytes commented on dftool:
```diff
@@ -590,6 +593,48 @@ ADDRESS_FILE -- eg: export ADDRESS_FILE={networkutil.chainIdToAddressFile(chainI
 
     from util import oceantestutil  # pylint: disable=import-outside-toplevel
```
I agree that imports should be on top.

I believe most of these changes came from making the project pylint & black compatible.
I think there might be a test or two that were importing from outside the top-level.

I'm reviewing and preparing this PR.
If I find any imports out of place, I'll restructure.

idiom-bytes commented on dftool:
```diff
@@ -590,6 +593,48 @@ ADDRESS_FILE -- eg: export ADDRESS_FILE={networkutil.chainIdToAddressFile(chainI
+        print(HELP)
+        sys.exit(0)
+
+    from util import oceantestutil  # pylint: disable=import-outside-toplevel
```
I moved it to the very top.
I believe I was following the same convention as `test_calcrewards.py`.

The reasoning for this (I believe) is to improve speed of tests & coverage.

For example, let's use the case of `test_calcrewards.py` and assume 2 testing branches:
1. partial tests
2. full tests

The partial run would be faster if it didn't have to load unused libraries at top level import.
Example, matplotlib (or other data) could be loaded just for delivering full test coverage.

In this case, I moved the oceantestutil imports to the top.

idiom-bytes commented on util/oceantestutil.py:
```diff
@@ -28,9 +28,28 @@
+    bal_before: int = fromBase18(token.balanceOf(account))
+
+    if bal_before < amount:
+        token.mint(account, toBase18(amount), {"from": account})
```
Hmm, I don't think so.
Let's say we want to init account[0].balance to 1,000, and the current balance is zero.
```
amount = 1000
bal_before: int = fromBase18(token.balanceOf(account))
assert bal_before == 0

if bal_before < amount :
   token.mint(account, toBase18(bal_before), {"from": account})
```
The user balance remains at zero.

```
amount = 1000
bal_before: int = fromBase18(token.balanceOf(account))
assert bal_before == 0

if bal_before < amount :
   token.mint(account, toBase18(amount), {"from": account})
```
The user balance is at 1000.

So, this should maintain user balances at a minimum of what's expected.

trentmc commented:
I see a PR, and no github issue.

Guys, we have to stop doing this.

There needs to be a github issue, that outlines a problem.

Then the PR needs to solve the problem. And that's it.

This helps in several ways: prioritize the issues (before there is work / PRs on it), and keeps the PRs properly-scoped to 👏 just 👏 solve 👏 the 👏 issue 👏 and 👏 that's 👏 all 👏. 

Thank you!

trentmc commented on dftool:
```diff
@@ -30,6 +30,14 @@ from util.oceanutil import (
     FeeDistributor,
 )
 
+from util.oceantestutil import (
```
:(

This PR is not doing "what's on the box". It looks like it's doing a bunch of different things.

Please do not do this.

One github issue should outline one problem.

One PR should solve that one issue.

A PR should not be a grab bag of a bunch of things.

I see here that there are 3-5 things going on. Please create 3-5 github issues. Create a PR for each of them. And let's go through them.

We can go through them quickly because the scope for each is tight. Therefore it will be easy to review and approve.

This PR, I definitely can't approve.

idiom-bytes commented:
**[WRT Ticket]**
This PR never got a ticket because it's referencing an issue inside of df-web => https://github.com/oceanprotocol/df-web/issues/255

Yes, I should have created a separate issues inside of df-py and linked to it.

**[WRT out-of-box]**
- If the ticket better described the two commands to implement, would that be sufficient?
     - I saw your feedback in the review. I will split it into different PRs.
- Some of these changes (like abstracting the ARGV table) only happened at the end of the implementation when pylint throws errors.

idiom-bytes commented:
You are right, there was some cruft in here... I cleaned things up.
Helper command initdevwallet => https://github.com/oceanprotocol/df-py/pull/498
Update manyrandom to latest code => https://github.com/oceanprotocol/df-py/pull/496
Cleaned up do_main for pylint => https://github.com/oceanprotocol/df-py/pull/494

trentmc commented:
I've approved all three of the new PRs. I'm fine if you close this PR then. (And delete related branch)
